src/controllers/follower.js

```diff
@@ -0,0 +1,110 @@
+import { User, Follower } from '../models';
+import { responseFormat, errorResponseFormat } from '../utils';
+
+
+/**
+ * @name FollowUser
+ * @description This is the method for indicationg a user to follow in the request body
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @returns {int} Returns the followed user
+ */
+export const followUser = async (req, res) => {
+  try {
+    const { user } = req;
+    const { followerId } = req.params;
+    const findUser = await User.findOne({ where: { id: followerId } });
+    if (!findUser) {
+      return res.status(404).json(
+        errorResponseFormat({
+          status: 'fail',
+          message: 'User not found!'
+        })
+      );
+    }
+    if (findUser.id === user.id) {
+      return res.status(403).json(
+        errorResponseFormat({
+          status: 'fail',
+          message: 'Sorry, you can not follow yourself'
+        })
+      );
+    }
+    const isFollowed = await Follower.findOrCreate({ where: { userId: user.id, followerId }
+    });
+    if (isFollowed[1] === false) {
+      return res.status(400).json(
+        errorResponseFormat({
+          status: 'fail',
+          message: 'You are already following this user'
+        })
+      );
+    }
+    return res.status(200).json(
+      responseFormat({
+        status: 'success',
+        data: 'You are now following this user'
+      })
+    );
+  } catch (error) {
+    return res.status(400).json(
+      errorResponseFormat({
+        status: 'fail',
+        message: 'Something went wrong, please try again later!'
+      })
+    );
+  }
+};
+
+/**
+ * @name unFollowUser
+ * @description This is the method for indicationg a user to follow in the request body
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @returns {int} Returns the followed user
+ */
+export const unfollowUser = async (req, res) => {
+  try {
+    const { user } = req;
+    const { followerId } = req.params;
+    const findUser = await User.findOne({ where: { id: followerId } });
+    if (!findUser) {
+      return res.status(404).json(
+        errorResponseFormat({
+          status: 'fail',
+          message: 'User not found!'
+        })
+      );
+    }
+    if (findUser.id === user.id) {
+      return res.status(403).json(
+        errorResponseFormat({
+          status: 'fail',
+          message: 'Sorry, you can not unfollow yourself'
+        })
+      );
```

Comment:
@Raymond-Osy you can implement follow and unfollow as a toggle functionality. click to follow. once you click again you unfollow that way you can use findOrCreate you already have for following functionality

Reply:
@Raymond-Osy. Nice work.  you can modify the error message to only return  `status` and `data` in the body.

Reply:
Thanks, @ssanusi for the feedback, I never thought of implementing it that way. I have implemented these changes.

Reply:
Thanks @micah-akpan for spotting it, I've adjusted it accordingly